src/lib/abacus/stateNotification.ts

```diff
@@ -154,33 +156,70 @@ class AbacusStateNotifier implements AbacusStateNotificationProtocol {
       }
 
       subaccountNumbers?.forEach((subaccountId: number) => {
-        if (subaccountId !== null) {
-          if (changes.has(Changes.subaccount)) {
-            dispatch(setSubaccount(updatedState.subaccount(subaccountId)));
+        const isChildSubaccount = subaccountId >= NUM_PARENT_SUBACCOUNTS;
+        const childSubaccountUpdate: Parameters<typeof setChildSubaccount>[0] = {};
+
+        if (changes.has(Changes.subaccount)) {
+          const subaccountData = updatedState.subaccount(subaccountId);
+          if (isChildSubaccount) {
+            childSubaccountUpdate[subaccountId] = subaccountData;
+          } else {
+            dispatch(setSubaccount(subaccountData));
           }
+        }
 
-          if (changes.has(Changes.fills)) {
-            const fills = updatedState.subaccountFills(subaccountId)?.toArray() ?? [];
+        if (changes.has(Changes.fills)) {
+          const fills = updatedState.subaccountFills(subaccountId)?.toArray() ?? [];
```

Comment:
hmm you're right. it might have something to do with the internal data in Abacus. We don't reference the `childSubaccount` struct yet, so I think this can be clarified as a follow up unless you are seeing some weirdness when this is missing.